RestApi/views.py

```diff
@@ -2,21 +2,42 @@
 from rest_framework.views import APIView
 from rest_framework.response import Response
 from rest_framework import status
-from .serializers import AnswerSerializer, TagSerializer, QuestionSerializer
+from .serializers import AnswerSerializer, TagSerializer, QuestionSerializer, UsernameSerializer
+from .models import UsernameSnippet
 from rest_framework.decorators import api_view
 from threads.models import answer
 from ask.models import tag, question
+from django.contrib.auth.models import User
+from django.core.exceptions import ObjectDoesNotExist
 
 
 def TestView(request):
-    return render(request, "test.html")
+    def get():
+        pass
 
 
 class TestPost(APIView):
     def post(self):
         pass
 
 
+class CheckUsername(APIView):
+    def post(self, request):
+        uname = request.data["username"]
+        try:
+            u = User.objects.get(username=uname)
```

Comment:
@digi0ps Can you ignore cases when checking for the username? Jeremy is available while jeremy already exists. Our current registration model only accepts the creation of an unique username (case insensitive).